src/main/java/uk/gov/pay/directdebit/mandate/params/MandateSearchParams.java

```diff
@@ -101,18 +107,47 @@ public MandateSearchParams withToDate(ZonedDateTime toDate) {
         return this;
     }
 
-    public MandateSearchParams withPage(int page) {
-        this.page = page;
+    public MandateSearchParams withPage(Integer page) {
+        if (page != null && page < 1) {
+            throw new BadRequestException("Query param 'page' should be a non zero positive integer.");
```

Comment:
This message (and others) matches what we have for Direct Debit payments but is slightly different from what we do in card connector (the message has a lower-case “q” for “query”). I don’t know how much this matters.